Marlin/planner.cpp

```diff
@@ -601,26 +591,47 @@ void Planner::check_axes_activity() {
 #endif // PLANNER_LEVELING
 
 /**
- * Planner::buffer_line
+ * Add a new linear movement to the buffer.
+ * The target is cartesian.
+ *
+ *  target   - x,y,z,e CARTESIAN target in mm
+ *  fr_mm_s  - (target) speed of the move (mm/s)
+ *  extruder - target extruder
+ */
+void Planner::buffer_line_kinematic(const float target[NUM_AXIS], float fr_mm_s, const uint8_t extruder) {
+  const float* t=target;
+  #if PLANNER_LEVELING
+    float leveled[NUM_AXIS];
+    memcpy(leveled, target, sizeof(leveled));
+    apply_leveling(leveled);
+    t=leveled;
+  #endif
```

Comment:
why? -obviously I had ask- why :-). Right. it's good idea to save stack or registers putting only reference when you not need to modify the value.

> But unfortunately, the compiler doesn't treat const & as immutable. 
this is strange, the compiler should raise error when you try to modify const variable or assign it to non const pointer. I've not try it, however maybe that is some problem of avr kind of compiler.

However, I've modified the functions *_kinematic to be FORCE_INLINE and to avoid memcpy. The assignment of pointer to const pointer is not problem, it's allowed operation, however I've modified it following your suggestions. Maybe that we save some assignment. I've not checked the resulting code, however I think, that it's optimal.